### PR TITLE
Transform Improvements

### PR DESCRIPTION
**Pull request by maheeka**

### CODE REVIEW

File: modules/web/js/ballerina/env/environment.js [hunk 1]
```diff
@@ -179,6 +182,25 @@ class BallerinaEnvironment extends EventChannel {
     }
 
     /**
+     * Initialize operator lattice
+     */
+    initializeOperatorLattice() {
+        return new Promise((resolve, reject) => {
```
Comment (roonyh) [inline modules/web/js/ballerina/env/environment.js]: No need to create a new promise here. You can return the promise received from `getOperatorLatice`.
~~~
return getOperatorLattice()
                 .then((operatorLatticeJson) => {
                     if (operatorLatticeJson) {
                         this._operatorLattice.initFromJson(operatorLatticeJson);
                     } else {
                         log.error('Error while fetching operator lattice');
                     }
                 })
~~~

File: modules/web/js/type-lattice/operator-lattice.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        const binaryLattice = {};
+
+        const unaryLatticeJson = operatorLatticeJson.filter(edge => edge.type === 'unary');
+        const binaryLatticeJson = operatorLatticeJson.filter(edge => edge.type === 'binary');
+
+        unaryLatticeJson.forEach((edge) => {
+            if (!edge.visited) {
+                const operator = edge.operator;
+                unaryLattice[operator] = {};
+                if (!edge.visited) {
```
Comment (roonyh) [inline modules/web/js/type-lattice/operator-lattice.js]: The check `if (!edge.visited)` is already done in line 47. No need to do it again right?  
Also we can do an early return,
`if (edge.visited) { return; }`

File: modules/web/js/type-lattice/operator-lattice.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    });
+                }
+            }
+        });
+
+        binaryLatticeJson.forEach((edge) => {
+            if (!edge.visited) {
+                const operator = edge.operator;
+                binaryLattice[operator] = {};
+                if (!edge.visited) {
```
Comment (roonyh) [inline modules/web/js/type-lattice/operator-lattice.js]: Same as above, no need to check again?

File: modules/web/js/ballerina/env/environment.js (hunk 1 above)
Comment (maheeka) [reply modules/web/js/ballerina/env/environment.js]: There are no promises returning from getOperatorLattice as we have hardcoded it for the moment. Shall we leave this for now? I'll add a TODO

File: modules/web/js/type-lattice/operator-lattice.js (hunk 2 above)
Comment (maheeka) [reply modules/web/js/type-lattice/operator-lattice.js]: This is a two level check. We are iterating over the same list twice and hence the two checks.

File: modules/web/js/type-lattice/operator-lattice.js (hunk 3 above)
Comment (maheeka) [reply modules/web/js/type-lattice/operator-lattice.js]: Same as above

File: modules/web/js/type-lattice/operator-lattice.js (hunk 2 above)
Comment (roonyh) [reply modules/web/js/type-lattice/operator-lattice.js]: um. but then the second if check should be inside the second forEach ne? These two ifs are in the same iteration. :thinking:

File: modules/web/js/ballerina/env/environment.js (hunk 1 above)
Comment (roonyh) [reply modules/web/js/ballerina/env/environment.js]: `Promise.resolve` returns a promise here https://github.com/ballerinalang/composer/pull/3335/files#diff-e30e1d6640d67749adfae49b49d69af0R293

**Pull request merged**